### DISCUSSION TITLE
Redis switch to SSPL - what does it mean for the redis-operator?

### DISCUSSION THREAD

**Discussion by ChristianHeich-bosch** [OT-CONTAINER-KIT/redis-operator#1405] [Q&A]
Hi all,

I am currently testing your amazing operator and it looks promising! Thanks for that :)

Now I have read that [Redis is changing it's license to the SSPL](https://redis.io/blog/redis-adopts-dual-source-available-licensing/) and would like to know if there are already some plans available regarding this operator. Are you switching to f. e. [Valkey](https://github.com/valkey-io/valkey) from the [Linux Foundation](https://www.linuxfoundation.org/press/linux-foundation-launches-open-source-valkey-community)?

Thanks and best regards,
Christian

**Comment by spron-in** [comment]
Hey folks. I'm from Percona and we are strong supporters of Valkey. Will you be open to accept contributions to support Valkey? I think this would be beneficial for the community in a long run.

**Comment by arpan57** [comment]
hi @spron-in - 

Do you have a fork/PR where we can see potential changes? It would help anyone who is trying to patch Redis Operators for making it work for ValKey. 

Thanks.

**Comment by pierreozoux** [comment]
@arpan57 you said you have it running with valkey, do you have anything to share maybe?

thanks!

**Comment by arpan57** [comment]
Hi @pierreozoux  - I might submit the changes/raise a potential PR in future. 
Good starting point is to replace the scripts with redis-cli to valkey-cli and change the CRD's groups. Search references and replace by new ones. 

Container docker file also needs to refer to the valkey image - based on the version.

**Comment by btoic** [comment]
Based on my internal testing, nothing at all needs to be changed on the operator side, you just need a Valkey image to use with the operator (OT-CONTAINER-KIT/redis entry points, etc...)

I've used the original [OT-CONTAINER-KIT/redis ](https://github.com/OT-CONTAINER-KIT/redis)image and modified it to use Valkey as base container, you can find the changes in this branch: [https://github.com/btoic/valkey/tree/import ](https://github.com/btoic/valkey/tree/import)

Valkey base container image maintains backwards compatibility with Redis by symlinking all the required binaries.

Tested this internally, both with Valkey 7.2.8 and 8.1.0 works fine in cluster mode. Images are available here: https://github.com/btoic/valkey/pkgs/container/valkey

While not ideal in terms of naming confusion that might arise (using redis-cli commands in Valkey images), for now it works well. In the long term, it would be nice to have CRD config setting the proper image variant and modifying the commands as required.

**Comment by pierreozoux** [comment]
Redis switched to AGPL :)

Not sure what to do with this issue?

Is there plans to support valkey or not? Maybe it should be a separate issue :)

**Comment by chris-gooch** [comment]
What’s the latest on this? Can we just use valkey images? Is the operator docs going to get updated to reflect this?

**Comment by odoucet** [comment]
Hello,
OT-Helm operator is fully compatible with Valkey images because valkey provides 'redis-*' symbolink links to valkey binaries.

I was able to make it work easily like this : 
1. If you do not need custom config, just override image with `ghcr.io/valkey-io/valkey` (and tag to `9-alpine` for example)
2. If you do need custom config (through `redisConfig` in CRD, or `externalConfig` in helm charts, you need to add an additional env value : 
```
env:
  - name: VALKEY_EXTRA_FLAGS
    value: "--include '/etc/redis/external.conf.d/*.conf'"
```

This is because by default, Redis reads this folder and Valkey does not. This ensures that any custom config is read by Valkey.
